homeassistant/components/dexcom/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""The Dexcom integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pydexcom import AccountError, Dexcom, SessionError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_UNIT_OF_MEASUREMENT, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_SERVER, DOMAIN, MG_DL, PLATFORMS, SERVER_OUS
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=180)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up configured Dexcom."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Dexcom from a config entry."""
+    try:
+        dexcom = await hass.async_add_executor_job(
+            Dexcom,
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            entry.data[CONF_SERVER] == SERVER_OUS,
+        )
+    except AccountError:
+        return False
+    except SessionError:
+        raise ConfigEntryNotReady
+
+    entry.update_listeners = []
```

Comment:
Regarding previous comment: an `update_listener` is added to the `config_entry`'s `update_listeners` every time the entry is loaded. This was why it failed on the *second* option change. So, while reloading the entry all listeners are removed. Let me know of any issues, or more suitable methods.

Reply:
Remove this line and instead store the remove callback that is returned when adding the update listener. Call the remove callback in `async_unload_entry`